src/robusta/patch/patch.py

```diff
@@ -0,0 +1,122 @@
+from typing import Union, List, Dict, get_type_hints
+from dataclasses import is_dataclass, InitVar
+from inspect import signature, getmodule
+from hikaru import HikaruDocumentBase, HikaruBase
+from ruamel.yaml import YAML
+
+try:
+    from typing import get_args, get_origin
+except ImportError:  # pragma: no cover
+    def get_args(tp):
+        return tp.__args__ if hasattr(tp, "__args__") else ()
+
+    def get_origin(tp):
+        return tp.__origin__ if hasattr(tp, "__origin__") else None
+
+NoneType = type(None)
+
+
+def create_monkey_patches():
```

Comment:
I would add a comment here explaining the motivation for each of these patches.

Reply:
Adding